examples/SimpleRagas/assets/example_setupcode.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+import os
+import re
+import tempfile
+import lancedb
+
+import braintrust
+import markdownify
+import openai
+import requests
+
+# We're only going to embed a small subset of our total data for this example, but you can increase this number to embed more sections!
+NUM_SECTIONS = 20
+EMBEDDING_MODEL = "text-embedding-3-small"
+
+
+coda_qa_content_data = requests.get(
+    "https://gist.githubusercontent.com/wong-codaio/b8ea0e087f800971ca5ec9eef617273e/raw/39f8bd2ebdecee485021e20f2c1d40fd649a4c77/articles.json"
+).json()
+
+markdown_sections = [
+    {"doc_id": row["id"], "markdown": section.strip()}
+    for row in coda_qa_content_data
+    for section in re.split(r"(.*\n=+\n)", markdownify.markdownify(row["body"]))
+    if section.strip() and not re.match(r".*\n=+\n", section)
+]
+
+openai_client = braintrust.wrap_openai(
+    openai.AsyncOpenAI(
+        base_url="https://braintrustproxy.com/v1",
+        default_headers={"x-bt-use-cache": "always"},
+        api_key=os.environ.get("OPENAI_API_KEY", "Your OPENAI_API_KEY here"),
+    )
+)
+
+LANCE_DB_PATH = os.path.join(tempfile.TemporaryDirectory().name, "docs-lancedb")
+
+@braintrust.traced
+async def embed_text(text):
+    response = await openai_client.embeddings.create(input=text, model=EMBEDDING_MODEL))
+    embedding = response.data[0].embedding
+    braintrust.current_span().log(
+        metrics={
+            "tokens": response.usage.total_tokens,
+            "prompt_tokens": response.usage.prompt_tokens,
+        },
+        metadata={"model": response.model},
+        input=text,
+        output=embedding,
+    )
+    return embedding
+
+# Create embeddings for all sections
+embeddings = asyncio.run(
+    asyncio.gather(*[embed_text(section["markdown"]) for section in markdown_sections[:NUM_SECTIONS]])
+)
+
+# Connect to database and insert data
+db = lancedb.connect(LANCE_DB_PATH)
+table = db.create_table(
+    "sections",
+    data=[
+        {"doc_id": row["doc_id"], "section_id": i, "vector": embedding}
+        for i, (row, embedding) in enumerate(
+            zip(markdown_sections[:NUM_SECTIONS], embeddings)
+        )
+    ],
+)
```

Comment:
I think this part should be in the notebook. Probably ok to have two sections:

* Populating the vector database (feel free to use numpy if you want :))
* Running a RAG query